clients/python/girder_client/__init__.py

```diff
@@ -146,14 +166,20 @@ def __init__(self, host=None, port=None, apiRoot=None, scheme=None, apiUrl=None,
             to pass 443 for the port
         :param cacheSettings: Settings to use with the diskcache library, or
             None to disable caching.
+        :param progressReporterCls: the progress reported class to instantiate.
```

Comment:
reporter*

Let's also add a `:type:` specifier that says the class must be a context manager, and also must have an `update` method that receives a quantity. It's unclear right now whether the quantity is the current amount or an incremental amount; maybe we could document that inside the _NoopProgressReporter class if that's easier.